api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -0,0 +1,179 @@
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import logging
+import os
+
+from .set_swupdate_repo import SetSWUpdateRepo
+from .. import inputs, values
+from ..config import (REPO_CANDIDATE_NAME, SW_UPGRADE_REPOS, YUM_REPO_TYPE,
+                      PRVSNR_USER_FILES_SWUPGRADE_REPOS_DIR, RELEASE_INFO_FILE,
+                      ReleaseInfo
+                      )
+from ..errors import SaltCmdResultError, SWUpdateRepoSourceError
+from ..utils import load_yaml
+
+logger = logging.getLogger(__name__)
+
+
+class SetSWUpgradeRepo(SetSWUpdateRepo):
+
+    _REPO_DEST = PRVSNR_USER_FILES_SWUPGRADE_REPOS_DIR
+    _BASE_DIR_PILLAR = "release/upgrade/base_dir"
+
+    def _single_repo_validation(self, release, repo_name):
+        """
+        Separate private method for basic single repo validations.
+
+        Parameters
+        ----------
+        release:
+            Single Upgrade ISO release version
+        repo_name: str
+            Repository name
+
+        Returns
+        -------
+
+        """
+        logger.info(f"Validate single SW upgrade repo '{repo_name}' "
+                    f"of release {release}")
+
+        # general check from pkg manager point of view
+        try:
+            self._check_repo_is_valid(release, f"sw_upgrade_{repo_name}")
+        except SaltCmdResultError as exc:
+            raise SWUpdateRepoSourceError(
+                                        repo_name, f"malformed repo: '{exc}'")
+
+        # there is no the same release repo is already active
+        if self._is_repo_enabled(f'sw_update_{release}'):
+            err_msg = (
+                "SW update repository for the release "
+                f"'{release}' has been already enabled"
+            )
+            logger.warning(err_msg)
+
+            # TODO IMPROVE later raise and error
+            if False:
+                raise SWUpdateRepoSourceError(
+                    str(repo.source),
+                    (
+                        f"SW update repository for the release "
+                        f"'{release}' has been already enabled"
+                    )
+                )
+
+    def dynamic_validation(self, params: inputs.SWUpgradeRepo, targets: str):  # noqa: C901, E501
+        """
+        Validate single SW upgrade ISO structure.
+
+        Parameters
+        ----------
+        params: inputs.SWUpgradeRepo
+            Input repository parameters
+        targets: str
+            Salt target to perform base mount and validation logic
+
+        Returns
+        -------
+
+        """
+        repo = params
+
+        if repo.is_special():
+            logger.info("Skipping update repo validation for special value: "
+                        f"{repo.source}")
+            return
+
+        logger.info(f"Validating upgrade repo: release {repo.release}, "
+                    f"source {repo.source}")
+
+        candidate_repo = inputs.SWUpgradeRepo(REPO_CANDIDATE_NAME, repo.source)
```

Comment:
Can we have SWUpgradeRepo use `repo.source` from pillar rather than a parameter?
The reason being, a parameter would allow user to inject error, which we might not be always able to catch.
Moreover, it would be good to treat pillar data as single source of truth, to avoid any possible conflict situation.

Reply:
pillar is not aware about a new source for that moment, we put that into pillar at the end of that command logic